packages/health-check/src/checks/critical.js

```diff
@@ -7,53 +7,86 @@ const { calculateElapsedTime, getResponseContent } = require("../utils");
 async function uploadCheck(done) {
   const time = process.hrtime();
   const form = new FormData();
-  const data = Buffer.from(new Date()); // current date to ensure data uniqueness
+  const payload = Buffer.from(new Date()); // current date to ensure data uniqueness
+  const data = { up: false };
 
-  form.append("file", data, { filename: "time.txt", contentType: "text/plain" });
-  let statusCode, errorResponseContent;
+  form.append("file", payload, { filename: "time.txt", contentType: "text/plain" });
 
   try {
-    const response = await got.post(`${process.env.PORTAL_URL}/skynet/skyfile`, { body: form });
+    const response = await got.post(`${process.env.SKYNET_PORTAL_API}/skynet/skyfile`, { body: form });
```

Comment:
This seems to depend on the custom `/etc/hosts` that this PR is adding. Is this better than using `PORTAL_NAME`? It seems like an easy thing to mess up in the future if we decide to, for example, have the portal API be at `portal.siasky.net` and don't add that to the custom `/etc/hosts`. And then we'd end up with health checks going through the load balancer and being meaningless.

I don't mind the way it works right now but I'm afraid that this change is dangerous because it's setting us up for failure if we forget to do a seemingly unrelated change. The definition of tech debt.

Reply:
To ensure that we're not going through load balancer we have now the `ip` property that we can double check in case we're not sure - of course it's manual but that's additional layer of security.

PORTAL_NAME is not a good choice here because it can be anything, readme says:
> `PORTAL_NAME` a string representing name of your portal e.g. `siasky.xyz` or `my skynet portal`

I am aware of the potential issue with hardcoding account.siasky.net but there wasn't really easy way around it that wouldn't require adding a new env variable to every server. I would like to follow up at some point with a cleanup to .env file and all environment variables, this could also mean we would add specific env variable that we could use here.

Reply:
Right, I forgot that was a very open var.